src/dnvm/DnvmConfig.cs

```diff
@@ -0,0 +1,164 @@
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading.Tasks;
+using Serde;
+using Serde.Json;
+using Zio;
+using Zio.FileSystems;
+
+namespace Dnvm;
+
+/// <summary>
+/// Represents the dnvm configuration file.
+/// </summary>
+[GenerateSerde]
+public sealed partial record DnvmConfig
+{
+    public static readonly DnvmConfig Default = new();
+
+    /// <summary>
+    /// Whether to enable dnvm preview releases in the update command.
+    /// </summary>
+    public bool PreviewsEnabled { get; init; } = false;
+}
+
+/// <summary>
+/// Provides access to the dnvm configuration file.
+/// </summary>
+public sealed class DnvmConfigFile
+{
+    private const string ConfigFileName = "dnvmConfig.json";
+    
+    // Allow tests to override the config directory
+    public static string? TestConfigDirectory { get; set; }
+
+    private readonly string _configDirectory;
+
+    public DnvmConfigFile() : this(GetConfigDirectory())
+    {
+    }
+
+    internal DnvmConfigFile(string configDirectory)
+    {
+        _configDirectory = configDirectory;
+    }
+
+    /// <summary>
+    /// Get the platform-specific config directory path.
+    /// - Linux: ~/.config/dnvm/ (XDG_CONFIG_HOME/dnvm)
+    /// - macOS: ~/.config/dnvm/
+    /// - Windows: %APPDATA%/dnvm/
+    /// </summary>
+    private static string GetConfigDirectory()
+    {
+        if (TestConfigDirectory is not null)
+        {
+            return TestConfigDirectory;
+        }
+
+        // Allow tests to override config directory via environment variable
+        var testOverride = Environment.GetEnvironmentVariable("DNVM_TEST_CONFIG_DIR");
```

Comment:
I don't think we should have an override environment variable. Let's remove this.